setup.py

```diff
@@ -14,6 +14,7 @@
                 'partd >= 0.3.8', 'cloudpickle >= 0.2.1'],
   'distributed': ['distributed >= 1.21'],
   'delayed': ['toolz >= 0.7.3'],
+  'scipy': ['scipy >= 0.16.0'],
```

Comment:
What if we renamed `scipy` to `misc` or similar?

Reply:
It's not clear to me where to draw the line here.  

As an extreme example do include `keras` for distributed because we support optional serialization of Keras models within the codebase?  I think that we really shouldn't bother in this case.  SciPy is obviously more central, but it's not clear to me that it's central enough.

Reply:
If this hasn't shown itself to be a major issue then I'm inclined to wait until it becomes one.

If we feel a strong need to resolve this anyway then I suggest that we handle it with informative error messages when we import optional dependencies.

```python
scipy = safely_import('scipy', '>= 0.16.0')
```

Reply:
I don't really understand the pushback here.

> As an extreme example do include `keras` for distributed because we support optional serialization of Keras models within the codebase?

This feels like a logical fallacy to me (e.g. slippery slope and/or straw man).

> If this hasn't shown itself to be a major issue then I'm inclined to wait until it becomes one.

Why wait for this to be a problem when it is trivially solved today? No extra effort is spent merging this or renaming to `misc` (or similar).

> If we feel a strong need to resolve this anyway then I suggest that we handle it with informative error messages when we import optional dependencies

This wouldn't have solved the issue we had with `dask`'s soft dependency on `numpy` nor would it with `scipy`. For some context, please see [this comment]( https://github.com/pyFFTW/pyFFTW/pull/214#issuecomment-372737534 ) and this commit ( https://github.com/pyfftw/pyFFTW/commit/b8c4dc88a595e7309ec084273a908057b663f433 ).

Reply:
To me slippery slope is a genuine issue, not a logical fallacy, I might understand correctly though.

I'm fine with recording the restrictions of optional dependencies.  Including it here seems odd because I would never expect anyone to actually `pip install dask[misc]` nor would I recommend them to.  Perhaps just a throwaway dict somewhere?

I also don't think that we should do this for just scipy. Keras was an extreme example, but fastparquet, pyarrow, s3fs, gcsfs, requests, moto, boto3, hdfs3, and others are possibly at least as important as the use of scipy if aggregated across all dask/dask users.

Reply:
Depending on how slippery slope is used it can be an argument or it can be a fallacy (this is debated). The comment about fallacy was in regards to it being used with an (acknowledged) extreme end point. Made this comment only with the interest of avoiding my proposal from being misconstrued.

Yeah that makes sense. Still would like to have some way of asking for the `scipy` version constraint to be satisfied though. This ends up being useful downstream. For instance, [this is how we constrain `dask` currently]( https://github.com/pyFFTW/pyFFTW/blob/db78c2bc907ccd971091fb96bf1bb0d809990b4f/setup.py#L763 ). IDK what to call this though. `dask[array_complete]`?

Sure I have no problem with people adding other groups of optional dependencies. That said, I'm not really familiar with those and can't really advise how they should be handled except at the most general level. If someone would like to advise me what to add, I can add them. However if that is unclear, maybe someone more knowledgeable about how those dependencies should be used/shared could pick that task up.

Reply:
`dask[array_complete]` seems tricky because it could conceivably contain keras (complete is a hard word).

Reply:
Also thought about `dask[array_extended]`. Anyways naming is definitely hard. The point is some sort of indication this builds off of `dask[array]` and provides a few related things.